style: apply /style-guide pass to models/reports (#2687)
wandb/docs: models/reports/create-a-report.mdx

---
description: Create a W&B Report with the W&B App or programmatically.
title: Create a report
keywords: ["create report", "wandb-workspaces", "Report API", "programmatic reports"]
---

<Note>
W&B Report and Workspace API is in Public Preview.
</Note>

Select a tab to learn how to create a report in the W&B App or programmatically with the Report and Workspace API.

For an example of how to programmatically create a report, see this [Google Colab](https://colab.research.google.com/github/wandb/examples/blob/master/colabs/intro/Report_API_Quickstart.ipynb).

<Tabs>
<Tab title="W&B App">
1. Navigate to your project workspace in the W&B App.
2. In the upper-right corner of your workspace, click **Create report**.

    <Frame>
    <img src="/images/reports/create_a_report_button.png" alt="Create report button" />
    </Frame>

> 3. A modal appears. Select the charts you want to start with. You can add or delete charts later from the report interface.

Reviewer: Maybe I lost state on this but I thought we favored "drawer" over "modal".

Author: Kept `modal` here: the chart-selection popup is a centered modal (it's even how the screenshot is filename'd — `create_a_report_modal.png`), and in the rest of the repo we reserve `drawer` for slide-out side panels (settings drawers, sort/group drawers, etc., for example in `models/runs/compare-runs.mdx` and `models/runs/grouping.mdx`). Happy to switch if you've seen the UI flip to a side drawer.

    <Frame>
    <img src="/images/reports/create_a_report_modal.png" alt="Create report modal" />
    </Frame>

> 4. Select the **Filter run sets** option to prevent new runs from being added to your report. You can toggle this option on or off. After you click **Create report**, a draft report is available in the report tab to continue working on.

Reviewer: ```suggestion
4. Select the **Filter run sets** option to prevent new runs from being added to your report. You can toggle this option on or off. A draft report is saved automatically. Access it in the **Reports** tab.
```

    ```bash
    pip install wandb wandb-workspaces
    ```

> 2. Import workspaces.

Reviewer: Nit: That's not really what you're importing in this code.

    ```python
    import wandb
    import wandb_workspaces.reports.v2 as wr
    ```
3. Create a report with `wandb_workspaces.reports.v2.Report`. Create a report instance with the Report Class Public API ([`wandb.apis.reports`](/models/ref/python/public-api/api#reports)). Specify a name for the project.
    ```python
    report = wr.Report(project="report_standard")
    ```

> 4. Save the report. Reports aren't uploaded to the W&B server until you call the `.save()` method:

Reviewer: ```suggestion
4. Save the report. Reports aren't uploaded to W&B until you call the `.save()` method:
```
Does it support context managers?

Author: Applied the suggestion. On context managers — `wr.Report` doesn't currently implement `__enter__`/`__exit__` (checked [interface.py](https://github.com/wandb/wandb-workspaces/blob/main/wandb_workspaces/reports/v2/interface.py), `class Report(Base)` only exposes `save()`), so I left the `report = wr.Report(...)` / `report.save()` pattern as-is. If wandb-workspaces adds context-manager support later we can revisit.

    ```python
    report.save()
    ```

> For more information about how to edit a report interactively with the App UI or programmatically, see [Edit a report](/models/reports/edit-a-report/).

Reviewer: ```suggestion
For more information, see [Edit a report](/models/reports/edit-a-report).
```
More trailing `/`. I wonder if the docs skill can be augmented to fix this.
